### PR TITLE
[cli] Several changes

### LINKED ISSUE #1377: Validators should be able to deaffiliate through the CLI
### Expected Behavior

Validators should be able to deaffiliate (i.e. remove themself from all groups) through the CLI

### Current Behavior

Validators cannot deaffiliate through the CLI.

### PR DESCRIPTION
**Pull request by mcortesi**
### Description

Changes: 
 - validator deregister. 
 - Split affliliate/deaffiliate
 - UX: Add checks before doing a tx
 - Test: Support for testing commands
 - Add a few tests (we need much more)
 - Add Support to call validator action using authorized signer

### Related issues

- Fixes #1377 #1378 

### Backwards compatibility

Changes cli affiliate command

### CODE REVIEW

**Review by nategraf — APPROVED** [review]

File: packages/cli/src/commands/validatorgroup/member.ts [hunk 1]
```diff
@@ -42,22 +43,24 @@ export default class ValidatorGroupMembers extends BaseCommand {
...
+      .isSignerOrAccount()
+      .canSignValidatorTxs()
+      .signerAccountIsValidatorGroup()
+      .isValidator(res.args.validatorAddress)
+      .runChecks()
+
+    const validatorGroup = await validators.signerToAccount(res.flags.from)
     if (res.flags.accept) {
-      const tx = await validators.addMember(res.flags.from, (res.args as any).validatorAddress)
+      const tx = await validators.addMember(validatorGroup, res.args.validatorAddress)
```
Comment (asaj) [inline packages/cli/src/commands/validatorgroup/member.ts:56]: What populates the "from" here (and elsewhere) if we don't set it explicitly?

File: packages/cli/src/test-utils/ganache-test.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
+import Web3 from 'web3'
```
Comment (asaj) [inline packages/cli/src/test-utils/ganache-test.ts:1]: This looks familiar, is this the same as what we do in `contractkit`? If so, should we maybe create a `testing` package?

Doesn't need to be part of this PR

File: packages/cli/src/utils/checks.ts [hunk 3]
```diff
@@ -0,0 +1,163 @@
+import { Address } from '@celo/contractkit'
```
Comment (asaj) [inline packages/cli/src/utils/checks.ts:1]: Love this 👏

File: packages/contractkit/src/wrappers/LockedGold.ts [hunk 4]
```diff
@@ -118,12 +126,30 @@ export class LockedGoldWrapper extends BaseWrapper<LockedGold> {
   getValidatorFromAccount: (account: string) => Promise<Address> = proxyCall(
     this.contract.methods.getValidatorFromAccount
   )
+
+  /**
+   * Returns the account address given the signer for validating
+   * @param signer Address that is authorized to sign the tx as validator
+   * @return The Account address
+   */
+  validatorSignerToAccount: (signer: Address) => Promise<Address> = proxyCall(
```
Comment (asaj) [inline packages/contractkit/src/wrappers/LockedGold.ts]: Is this consistent with what @nambrot is renaming things? If not, can we aim for consistency?

File: packages/contractkit/src/wrappers/LockedGold.ts [hunk 5]
```diff
@@ -118,12 +126,30 @@ export class LockedGoldWrapper extends BaseWrapper<LockedGold> {
...
    * @return Returns `true` if account exists. Returns `false` otherwise.
    */
   isAccount: (account: string) => Promise<boolean> = proxyCall(this.contract.methods.isAccount)
+
+  /**
+   * Check if an address is a signer address
+   * @param address The address of the account
+   * @return Returns `true` if account exists. Returns `false` otherwise.
+   */
+  isSigner: (address: string) => Promise<boolean> = proxyCall(this.contract.methods.isAuthorized)
```
Comment (asaj) [inline packages/contractkit/src/wrappers/LockedGold.ts]: Ditto

**Review by asaj — APPROVED** [review]
This is great

File: packages/cli/src/test-utils/ganache-test.ts (hunk 2 above)
Comment (mcortesi) [reply packages/cli/src/test-utils/ganache-test.ts:1]: yes, it's the same, it's a good follow up pr

File: packages/contractkit/src/wrappers/LockedGold.ts (hunk 4 above)
Comment (mcortesi) [reply packages/contractkit/src/wrappers/LockedGold.ts]: i wanted to be on top of nam pr but couldn't...
so when we have that, we'll have to move some methods and have a consistent naming

File: packages/cli/src/commands/validatorgroup/member.ts (hunk 1 above)
Comment (mcortesi) [reply packages/cli/src/commands/validatorgroup/member.ts:56]: this line:
```ts
this.kit.defaultAccount = res.flags.from
```

File: packages/cli/src/utils/checks.ts (hunk 3 above)
Comment (mcortesi) [reply packages/cli/src/utils/checks.ts:1]: Thanks!

File: packages/contractkit/src/wrappers/LockedGold.ts (hunk 5 above)
Comment (mcortesi) [reply packages/contractkit/src/wrappers/LockedGold.ts]: answered above

File: packages/cli/src/commands/lockedgold/authorize.test.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import Web3 from 'web3'
+import { testWithGanache } from '../../test-utils/ganache-test'
+import Authorize from './authorize'
+import Register from './register'
+
+process.env.NO_SYNCCHECK = 'true'
+
+testWithGanache('account:authorize cmd', (web3: Web3) => {
```
Comment (yorhodes) [inline packages/cli/src/commands/lockedgold/authorize.test.ts]: awesome!

File: packages/cli/src/utils/command.ts [hunk 7]
```diff
@@ -1,30 +1,30 @@
 import { flags } from '@oclif/command'
+import { CLIError } from '@oclif/errors'
```
Comment (yorhodes) [inline packages/cli/src/utils/command.ts:2]: +1

**Review by yorhodes — APPROVED** [review]
LGTM

**Pull request merged**